### PR TITLE
Cleanup undo

### PR DESCRIPTION
**Pull request by abi87**
Moved all sidechain related stuff into ad-hoc structure in blockUndo

### CODE REVIEW

File: src/undo.h [hunk 1]
```diff
@@ -188,39 +123,165 @@ class CTxUndo
...
+        res += strprintf("lowQualityBwts.size %u\n", lowQualityBwts.size());
+        for(const auto& voidCertUndo: lowQualityBwts)
+            res += voidCertUndo.ToString() + "\n";
+
+        return res;
     }
 };
 
+inline CSidechainUndoData::AvailableSections operator | (CSidechainUndoData::AvailableSections lhs, CSidechainUndoData::AvailableSections rhs)
+{
```
Comment (alsala) [inline src/undo.h:275]: If the bitmask is typed as an uint8_t (as per comment above) we can remove these operators

**Review by alsala — CHANGES_REQUESTED** [review]

File: src/undo.h (hunk 1 above)
Comment (abi87) [reply src/undo.h]: Done

File: src/undo.h [hunk 2]
```diff
@@ -188,39 +123,165 @@ class CTxUndo
...
+    uint32_t sidechainUndoDataVersion;
+    enum AvailableSections : uint8_t
+    {
+        UNDEFINED               = 0,
+        SIDECHAIN_STATE         = 1,
+        MATURED_AMOUNTS         = 2,
+        LOW_QUALITY_CERT_DATA   = 4,
+        CEASED_CERTIFICATE_DATA = 8
+    };
+    AvailableSections contentBitMask;
```
[the remark this answers is not in the document]
Comment (abi87) [reply src/undo.h]: Agree, done

File: src/undo.h [hunk 3]
```diff
@@ -188,39 +123,165 @@ class CTxUndo
...
     }
 
     std::string ToString() const
     {
-        std::string str;
-        str += strprintf("  immAmount %d.%08d\n", appliedMaturedAmount / COIN, appliedMaturedAmount % COIN);
-        return str;
+        std::string res;
+        res += strprintf("contentBitMask=%u\n", contentBitMask);
+        if (contentBitMask && AvailableSections::SIDECHAIN_STATE)
```
[the remark this answers is not in the document]
Comment (abi87) [reply src/undo.h]: Done

**Pull request merged**